idea/src/org/jetbrains/kotlin/idea/intentions/ReplaceSingleLineLetIntention.kt

```diff
@@ -102,6 +103,7 @@ class ReplaceSingleLineLetIntention : SelfTargetingOffsetIndependentIntention<Kt
         return when (right) {
             is KtNameReferenceExpression -> right.text != parameterName
             is KtDotQualifiedExpression -> !right.hasLambdaExpression() && !right.nameUsed(parameterName)
+            is KtBinaryExpression -> right.operationToken == KtTokens.RANGE
```

Comment:
Don't quite understand why only `RANGE` is allowed here. Please consider other binary operators. It seems to me (may be I miss something) that it should be allowed for most of them.

Reply:
I wasn't too sure which case can be like this format.
```
Expression BinaryOperator KtBinaryExpression
```
In case, if there is unexpected syntax, it would break this intention.

---

I found the current implementation cannot catch this 

```
listOf(1,2,3).filter { it.let { it in IntRange(1, 10) } }
```

Let me fix this.